tests/unit/src/components/shared/AppFooter.spec.js

```diff
@@ -0,0 +1,142 @@
+import { shallowMount, createLocalVue } from '@vue/test-utils';
+import AppFooter from '@/components/shared/AppFooter.vue';
+
+describe('AppFooter.vue', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  beforeAll(() => {
+    Object.defineProperty(window, 'localStorage', {
+      value: {
+        getItem: jest.fn(),
+        writable: true
+      }
+    })
+  });
+
+  describe('data', () => {
+    it('should have "isBackToTopVisible" as false value', () => {
+      const componentState = AppFooter.data();
+      
+      expect(componentState.isBackToTopVisible).toEqual(false);
+    });
+
+    it('should have "isFooterSubscribeVisible" as true value if "subscribeClosed" value is falsey', () => {
+      const componentState = AppFooter.data();
+      
+      expect(componentState.isFooterSubscribeVisible).toEqual(true);
+    });
+
+    it('should have "isFooterSubscribeVisible" as true value if "subscribeClosed" value is falsey', () => {
+      localStorage.getItem.mockReturnValue(false);
+      
+      const componentState = AppFooter.data();
+      
+      expect(componentState.isFooterSubscribeVisible).toEqual(true);
+    });
+
+    it('should have "isFooterSubscribeVisible" as false value if "subscribeClosed" value is true', () => {
+      localStorage.getItem.mockReturnValue(true);
+      
+      const componentState = AppFooter.data();
+      
+      expect(componentState.isFooterSubscribeVisible).toEqual(false);
+    });
+  });
+
+  describe('methods', () => {
+    describe('handleScroll', () => {
+      const setScrollY = (value) => {
+        Object.defineProperty(window, 'scrollY', {
+          value: value,
+          writable: true
+        });
+      };
+
+      it('assign "isBackToTopVisible" to true, if scrollY is greater than 540', () => {
+        const wrapper = shallowMount(AppFooter, {
+          stubs: ['router-link'],
+        });
+
+        setScrollY(541);
+
+        wrapper.vm.handleScroll();
+
+        expect(wrapper.vm.isBackToTopVisible).toEqual(true);
+      });
+
+      it('assign "isBackToTopVisible" to false, if scrollY is smaller than 540', () => {
+        const wrapper = shallowMount(AppFooter, {
+          stubs: ['router-link'],
+        });
+
+        setScrollY(539);
+
+        wrapper.vm.handleScroll();
+
+        expect(wrapper.vm.isBackToTopVisible).toEqual(false);
+      });
+
+      it('assign "isBackToTopVisible" to false, if scrollY is equal than 540', () => {
+        const wrapper = shallowMount(AppFooter, {
+          stubs: ['router-link'],
+        });
+
+        setScrollY(540);
+
+        wrapper.vm.handleScroll();
+
+        expect(wrapper.vm.isBackToTopVisible).toEqual(false);
+      });
+    });
+  });
+
+  describe('templates', () => {
+    const shallowAppFooter = () => {
```

Comment:
Why didn't you use `beforeEach` hook? Also, each component should be destroyed after each `describe` scope to prevent memory leak. You can use `afterEach` hook for it.